cdap-ui/app/cdap/components/DataPrep/Directives/MarkAsError/index.js

```diff
@@ -0,0 +1,332 @@
+import {execute} from 'components/DataPrep/store/DataPrepActionCreator';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+
+require('./MarkAsError.scss');
+
+const PREFIX = 'features.DataPrep.Directives.MarkAsError';
+const conditionsOptions = [
+  'EMPTY',
+  'TEXTEXACTLY',
+  'TEXTCONTAINS',
+  'TEXTSTARTSWITH',
+  'TEXTENDSWITH',
+  'TEXTREGEX',
+  'divider',
+  'CUSTOMCONDITION'
+];
+export default class MarkAsError extends Component {
+  state = {
+    selectedCondition: conditionsOptions[0],
+    customCondition: null,
+    textFilter: '',
+    customFilter: '',
+    ignoreCase: false
+  };
+
+  componentDidMount() {
+    this.calculateOffset = setPopoverOffset.bind(this, document.getElementById('mark-as-error-directive'));
+    this.offsetCalcDebounce = debounce(this.calculateOffset, 1000);
+  }
+
+  componentDidUpdate() {
+    if (this.props.isOpen && this.calculateOffset) {
+      this.calculateOffset();
+    }
+    if (this.state.selectedCondition.substr(0, 4) === 'TEXT' && this.state.textFilter.length === 0 && this.textFilterRef) {
+      this.textFilterRef.focus();
+    } else if (this.state.selectedCondition.substr(0, 6) === 'CUSTOM' && this.state.customFilter.length === 0 && this.customFilterRef) {
+      this.customFilterRef.focus();
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.offsetCalcDebounce);
+  }
+
+  applyDirective = () => {
+    execute([this.getDirectiveExpression()])
+      .subscribe(() => {
+        this.props.close();
+        this.props.onComplete();
+      }, (err) => {
+        console.log('error', err);
+
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setError,
+          payload: {
+            message: err.message || err.response.message
+          }
+        });
+      });
+  };
+
+  handleTextFilterChange = (e) => {
+    this.setState({textFilter: e.target.value});
+  };
+
+  handleCustomFilterChange = (e) => {
+    this.setState({customFilter: e.target.value});
+  };
+
+  toggleIgnoreCase = () => {
+    this.setState({
+      ignoreCase: !this.state.ignoreCase
+    });
+  };
+
+  getDirectiveExpression = () => {
+    let directive;
+    let condition = 'send-to-error';
+    let column = this.props.column;
+    let textValue = this.state.textFilter;
+    let configuration;
+
+    switch (this.state.selectedCondition) {
+      case 'EMPTY':
+        directive = `${condition} empty(${column})`;
+        break;
+      case 'TEXTCONTAINS':
+        if (this.state.ignoreCase) {
+          textValue = `(?i)${textValue}`;
+        }
+        directive = `${condition} "${column}" =~ "${textValue}.*"`;
+        break;
+      case 'TEXTSTARTSWITH':
+        configuration = `"${textValue}"`;
+        if (this.state.ignoreCase) {
+          column = `${column}.toLowerCase()`;
+          configuration = `"${textValue}".toLowerCase()`;
```

Comment:
These can be simplified to `column = column.toLowerCase()` and `configuration = configuration.toLowerCase()` right? Same in the `TEXTENDSWITH` case